oximeter/collector/src/agent.rs

```diff
@@ -496,6 +507,7 @@ impl OximeterAgent {
         let log = log.new(o!(
             "component" => "oximeter-standalone",
             "collector_id" => id.to_string(),
+            "collector_ip" => address.ip().to_string(),
```

Comment:
I _think_ you can write this as

```suggestion
            "collector_ip" => %address.ip(),
```

where the `%` prefix is allowed on anything that impls `Display`. It's possible that `slog` takes this and just calls `.to_string()` on it under the hood, but in theory they could do something smarter here, I think, and not allocate the string until it's needed (or maybe at all? I'd have to go learn more about the internals).

Reply:
Here's what you get with your change:

```
error[E0277]: `core::fmt::rt::Opaque` cannot be shared between threads safely
    --> oximeter/collector/src/agent.rs:507:27
     |
507  |           let log = log.new(o!(
     |  ___________________________^
508  | |             "component" => "oximeter-standalone",
509  | |             "collector_id" => id.to_string(),
510  | |             "collector_ip" => %address.ip(),
511  | |         ));
     | |         ^
     | |         |
     | |_________`core::fmt::rt::Opaque` cannot be shared between threads safely
     |           required by a bound introduced by this call
     |
     = help: within `[core::fmt::rt::Argument<'_>]`, the trait `Sync` is not implemented for `core::fmt::rt::Opaque`
     = note: required because it appears within the type `&core::fmt::rt::Opaque`
note: required because it appears within the type `core::fmt::rt::Argument<'_>`
    --> /home/bnaecker/.rustup/toolchains/1.76.0-x86_64-unknown-illumos/lib/rustlib/src/rust/library/core/src/fmt/rt.rs:75:12
     |
75   | pub struct Argument<'a> {
     |            ^^^^^^^^
     = note: required because it appears within the type `[core::fmt::rt::Argument<'_>]`
     = note: required for `&[core::fmt::rt::Argument<'_>]` to implement `std::marker::Send`
note: required because it appears within the type `Arguments<'_>`
    --> /home/bnaecker/.rustup/toolchains/1.76.0-x86_64-unknown-illumos/lib/rustlib/src/rust/library/core/src/fmt/mod.rs:303:12
     |
303  | pub struct Arguments<'a> {
     |            ^^^^^^^^^
note: required because it appears within the type `SingleKV<Arguments<'_>>`
    --> /home/bnaecker/.cargo/registry/src/index.crates.io-6f17d22bba15001f/slog-2.7.0/src/lib.rs:3284:12
     |
3284 | pub struct SingleKV<V>(pub Key, pub V)
     |            ^^^^^^^^
     = note: required because it appears within the type `(SingleKV<Arguments<'_>>, (SingleKV<std::string::String>, (SingleKV<&str>, ())))`
     = note: required for `(SingleKV<Arguments<'_>>, (SingleKV<std::string::String>, (SingleKV<&str>, ())))` to implement `SendSyncRefUnwindSafeKV`
note: required by a bound in `OwnedKV`
    --> /home/bnaecker/.cargo/registry/src/index.crates.io-6f17d22bba15001f/slog-2.7.0/src/lib.rs:3408:8
     |
3400 | pub struct OwnedKV<T>(
     |            ------- required by a bound in this tuple struct
...
3408 |     T: SendSyncRefUnwindSafeKV + ?Sized;
     |        ^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `OwnedKV`
     = note: this error originates in the macro `o` (in Nightly builds, run with -Z macro-backtrace for more info)

```

Reply:
Woof, alright, sorry for the diversion. I guess that makes sense if slog is doing the exact optimization I hope it does (i.e., borrowing the value instead of `.to_string()`'ing it immediately).